lib/Feature/FeatureModelParser.cpp

```diff
@@ -249,29 +250,242 @@ bool FeatureModelSxfmParser::parseFeatureTree(xmlChar *FeatureTree) {
   {
     std::stringstream Ss(reinterpret_cast<const char *>(FeatureTree));
     std::string To;
+    string Name;
+    bool Opt;
+    int LastIndentation = -1;
+    int OrGroupCounter = 0;
+    std::map<int, string> IndentationToParentMapping;
```

Comment:
Why do you remember different indentations ? and what happens if you have two indentations that are on the same level but have different parents?

Isn't the current indentation level sufficient?

Reply:
We do have to remember the different indentations for cases such as:
```
:r root (root_id)
  :o opt1 (id_opt1)
    :o opt2 (id_opt2)
      :o opt3 (id_opt3)
  :o opt4 (id_opt4)
```
The indentation between op3 and opt4 is -2 and the parent of opt4 is root and not opt2.
Thus, it does not suffice to only remember the last parent. 

> and what happens if you have two indentations that are on the same level but have different parents?

Should be no problem since a new parent of a certain indentation level overwrites the last one. There is no case where we would still need the last parent.